lib/dkim.js

```
@@ -463,8 +464,9 @@ function DKIMVerify(email, callback, sigs) {
 
         // The other DNS errors are operational in nature;
         sigs.result = sig.result = false;
-        sigs.issue_desc = sig.issue_desc = err;
-        return callback(err, sigs);
+        sigs.issue_desc = sig.issue_desc = err.message;
+
+        return callback(null, sigs); // returning the error AND the sigs breaks promisification... should do callback(null, sigs)
```

returning both shouldn't have any effect on promisification? although i think maybe the issue is that the error is returned at all (ie i think `callback(err)` would break the same way as it does currently) ... so we intend to swallow all these errors is that correct?

---

right, it's returning the error that breaks promisification... that comment isn't totally clear :)

We don't have any logging in this lib, which is unfortunate. And we're exposing the error message as is, which might also be weird. The better fix might be to add msys-logger and log the error, and then return a hard-coded generic error message. What do you think?

---

Ha ok just so I understand -- does it break something in how promises work somehow or does it just continue to throw an error that we actually want to swallow? I think it's the latter which may just be a semantic difference in how we're describing this but wanted to make sure I understood what was happening. :)